lib/client/signalflow/request_manager.js

```diff
@@ -390,11 +390,47 @@ function RequestManager(options) {
     //destroy all open SSE connections
   }
 
+
+  function registerLiveTail(params, callbacks, overrideRequestId){
+    var requestId = overrideRequestId || getRequestIdAndIncrement();
+    knownComputations[requestId] = {
+      onMessage: callbacks.onMessage,
+      transport: transports.WEBSOCKET,
+      requestId: requestId,
+    };
+    knownComputations[requestId].streamController = getLiveTailStreamController();
```

Comment:
This won't work because your stream controller returned by `getLiveTailStreamController()` does not capture the arguments of the original request. If you look at https://github.com/signalfx/signalfx-nodejs/blob/master/lib/client/signalflow/request_manager.js#L75 and https://github.com/signalfx/signalfx-nodejs/blob/master/lib/client/signalflow/request_manager.js#L364 you'll see that `retry()` and `stop()` are called without arguments.

You need to capture them here with a closure IMHO:

```js
knownComputations[requestId] = {
  onMessage: callbacks.onMessage,
  transport: transports.WEBSOCKET,
  requestId: requestId,
  streamController: {
    retry: () => registerLiveTail(params, callbacks, overrideRequestId),
    stop: () => stopLiveTail(requestId),
  },
};
```

Reply:
got it. yep. In my case I actually have arguments and need to close over them!!

Reply:
fixed